### PR TITLE
LPS-71982 Add required file liferay-ddm-form-renderer-field-feedback

### PR DESCRIPTION
**Pull request by alinedoleron**

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test

**Comment by alinedoleron** [comment]
ci:test

**Comment by pedroqueiroz94** [comment]
Hey @alinedoleron , there are Source Formatting issues in your PR. 

Can you please fix and resend? Thank you!

**Comment by alinedoleron** [comment]
Hi @pedroqueiroz94, I've already done the formatSource but I forgot to comment above.

File: dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date.soy [hunk 1]
```diff
@@ -44,6 +44,7 @@
 				{if $readOnly}disabled{/if}
 				type="text"
 				value="{$displayValue}"
+				maxlength="10"
```
Comment (brunobasto) [inline dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date.soy]: Alphabetical order

File: dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date_field.js [hunk 2]
```diff
@@ -42,14 +42,26 @@ AUI.add(
 
 						instance._eventHandlers.push(
 							datePicker.after('selectionChange', A.bind('_afterSelectionChange', instance)),
-							datePicker.on('activeInputChange', A.bind('_onActiveInputChange', instance))
+							datePicker.on('activeInputChange', A.bind('_onActiveInputChange', instance)),
+							instance.bindContainerEvent('input', instance._onStringType, '.trigger'),
```
Comment (brunobasto) [inline dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date_field.js]: Consistency on event names. 

_onInput
_onFocusOut

File: dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date_field.js [hunk 3]
```diff
@@ -164,6 +214,41 @@ AUI.add(
...
+						var anySymbolRegex = /[-!$%^&*()_+|~=`{}\[\]:\\";'<>?,.]/;
+						var barRegex = /\//;
+						var stringRegex = /[a-zA-Z\u00C0-\u00FF ]+/i;
+
+						if (input == '') {
+							instance.hideError();
+						}
+						else if (input.match(stringRegex) || input.match(anySymbolRegex)) {
+							instance.setValue('');
+							instance.addErrorMessage('This field does not accept letters');
```
Comment (brunobasto) [inline dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date_field.js]: We should not add hard coded strings. Use Liferay.Language.get for that.

File: dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date_field.js [hunk 4]
```diff
@@ -164,6 +214,41 @@ AUI.add(
...
+						}
+						else {
+							instance.hideError();
+							if (input.length === 8 && !input.match(barRegex)) {
```
Comment (brunobasto) [inline dynamic-data-mapping-type-date/src/main/resources/META-INF/resources/date_field.js]: Why 8?
What happens in the case of 

02/02/2002 -> 10 characters
2/1/17 -> 6 characters
15/4/2017 -> 9
1/1/1989 -> 8

**Comment by alinedoleron** [comment]
ci:test

**Comment by pedroqueiroz94** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by pedroqueiroz94** [comment]
PR resent at https://github.com/liferay/com-liferay-dynamic-data-mapping/pull/596

**Comment by pedroqueiroz94** [comment]
ci:close

**Pull request closed without merging**